plone/event/recurrence.py

```diff
@@ -103,8 +103,20 @@ def recurrence_sequence_ical(
             recrule = re.sub(r"T000000", t0str, recrule)
         # Then, replace each until times with the end of the day
         recrule = re.sub(
-            r"(UNTIL[^T]*[0-9]{8})T([0-9]{6}Z?)",
-            r"\1T235959",
+            r"(UNTIL[^T]*[0-9]{8})T([0-9]{6})(Z?)",
+            r"\1T235959\3",
+            recrule,
+        )
+
+        # XXX: The time of EXDATE, according to previous comments, should be the same
+        #      as the start time.
```

Comment:
To give some context for future readers please add a link to the mentioned comments and maybe the link to the specification https://icalendar.org/iCalendar-RFC-5545/3-8-5-1-exception-date-times.html
This is a complex topic, any hint is appreciated.

Reply:
cite from spec: 
"The ["EXDATE"](https://icalendar.org/iCalendar-RFC-5545/3-8-5-1-exception-date-times.html) property can be used to exclude the value specified in ["DTSTART"](https://icalendar.org/iCalendar-RFC-5545/3-8-2-4-date-time-start.html). However, in such cases, the original ["DTSTART"](https://icalendar.org/iCalendar-RFC-5545/3-8-2-4-date-time-start.html) date MUST still be maintained by the calendaring and scheduling system because the original ["DTSTART"](https://icalendar.org/iCalendar-RFC-5545/3-8-2-4-date-time-start.html) value has inherent usage dependencies by other properties such as the ["RECURRENCE-ID"](https://icalendar.org/iCalendar-RFC-5545/3-8-4-4-recurrence-id.html)."